pkg/scheduler/scheduler.go

```diff
@@ -460,9 +462,21 @@ func (s *Scheduler) QuerierLoop(querier schedulerpb.SchedulerForQuerier_QuerierL
 
 		r := req.(*schedulerRequest)
 
-		s.queueDuration.Observe(time.Since(r.enqueueTime).Seconds())
+		reqEnqueueTime := time.Since(r.enqueueTime).Seconds()
+		s.queueDuration.Observe(reqEnqueueTime)
 		r.queueSpan.Finish()
 
+		reqTenantIDs, err := tenant.TenantIDsFromOrgID(r.userID)
+		if err != nil {
+			reqTenantIDs = []string{}
+		}
+		reqURL, err := url.PathUnescape(r.request.Url)
+		if err != nil {
+			reqURL = r.request.Url
+		}
+		level.Info(s.log).Log("msg", "querier request dequeued", "tenant_ids", strings.Join(reqTenantIDs, ", "),
+			"querier_id", querierID, "query_id", r.queryID, "request", reqURL, "enqueue_time (ms)", reqEnqueueTime*1000)
+
```

Comment:
some small suggestions
```suggestion
		queueTime := time.Since(r.enqueueTime)
		s.queueDuration.Observe(queueTime.Seconds())
		r.queueSpan.Finish()

		reqTenantIDs, err := tenant.TenantIDsFromOrgID(r.userID)
		if err != nil {
			reqTenantIDs = []string{}
		}
		reqURL, err := url.PathUnescape(r.request.Url)
		if err != nil {
			reqURL = r.request.Url
		}
		level.Info(s.log).Log("msg", "querier request dequeued", "tenant_ids", strings.Join(reqTenantIDs, ", "),
			"querier_id", querierID, "query_id", r.queryID, "request", reqURL, "queue_time", queueTime)

```

Reply:
Thank you @owen-d 🙏 but in the meantime, I updated the solution to log the enqueue time in metrics.go in commit: https://github.com/grafana/loki/pull/4949/commits/3b11ac8f487646d6e9cc6f01a019551611dd3428

I took into account your chance for the `queueTime` time duration 👍